cpp/tests/io/json_type_cast_test.cu

```diff
@@ -0,0 +1,156 @@
+  auto const type   = cudf::data_type{cudf::type_id::STRING};
+
+  auto in_valids = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return i != 4; });
+  std::vector<const char*> input_values{"this", "is", "null", "of", "", "strings", R"("null")"};
+  cudf::test::strings_column_wrapper input(input_values.begin(), input_values.end(), in_valids);
+
+  auto d_column = cudf::column_device_view::create(input);
+  rmm::device_uvector<thrust::pair<const char*, cudf::size_type>> svs(d_column->size(), stream);
+  thrust::transform(thrust::device,
+                    d_column->pair_begin<cudf::string_view, false>(),
+                    d_column->pair_end<cudf::string_view, false>(),
+                    svs.begin(),
+                    to_thrust_pair_fn{});
+
+  auto null_mask_it = no_nulls();
+  auto null_mask =
+    cudf::test::detail::make_null_mask(null_mask_it, null_mask_it + d_column->size());
+
+  auto str_col = cudf::io::json::experimental::parse_data(
+    svs.data(), svs.size(), type, std::move(null_mask), default_json_options().view(), stream, mr);
+
+  auto out_valids =
+    cudf::detail::make_counting_transform_iterator(0, [](auto i) { return i != 2 and i != 4; });
+  std::vector<const char*> expected_values{"this", "is", "", "of", "", "strings", "null"};
+  cudf::test::strings_column_wrapper expected(
+    expected_values.begin(), expected_values.end(), out_valids);
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(str_col->view(), expected);
+}
+
+TEST_F(JSONTypeCastTest, Int)
+{
+  auto const stream = cudf::default_stream_value;
+  auto mr           = rmm::mr::get_current_device_resource();
+  auto const type   = cudf::data_type{cudf::type_id::INT64};
+
+  cudf::test::strings_column_wrapper data({"1", "null", "3", "true", "5", "false"});
+  auto d_column = cudf::column_device_view::create(data);
+  rmm::device_uvector<thrust::pair<const char*, cudf::size_type>> svs(d_column->size(), stream);
+  thrust::transform(thrust::device,
+                    d_column->pair_begin<cudf::string_view, false>(),
+                    d_column->pair_end<cudf::string_view, false>(),
+                    svs.begin(),
+                    to_thrust_pair_fn{});
+
+  auto null_mask_it = no_nulls();
+  auto null_mask =
+    cudf::test::detail::make_null_mask(null_mask_it, null_mask_it + d_column->size());
+
+  auto col = cudf::io::json::experimental::parse_data(
+    svs.data(), svs.size(), type, std::move(null_mask), default_json_options().view(), stream, mr);
+
+  auto expected =
+    cudf::test::fixed_width_column_wrapper<int64_t>{{1, 2, 3, 1, 5, 0}, {1, 0, 1, 1, 1, 1}};
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(col->view(), expected);
+}
+
+TEST_F(JSONTypeCastTest, StringEscapes)
+{
+  auto const stream = cudf::default_stream_value;
+  auto mr           = rmm::mr::get_current_device_resource();
+  auto const type   = cudf::data_type{cudf::type_id::STRING};
+
+  cudf::test::strings_column_wrapper data({
+    R"("\uD83D\uDE80")",
```

Comment:
Do we need some tests for other escaped characters like `\n`, `\f`, ...? I might just be missing where those tests live.

Reply:
Makes sense. Added tests for remaining escape sequences 👍